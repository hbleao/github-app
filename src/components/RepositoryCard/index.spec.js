import React from 'react';
import { screen, render } from '@testing-library/react';
import user from '@testing-library/user-event';
import renderer from 'react-test-renderer';

import { RepositoryCard } from '.';

const makeSut = ({ content, onClick = () => null }) => {
  return render(<RepositoryCard content={content} onClick={onClick} />);
};

const content = <p>facebook</p>;

describe('RepositoryCard', () => {
  it('should be able to render component correctly', () => {
    makeSut({ content });
    const repositoryName = screen.getByText(/facebook/);
    expect(repositoryName).toBeTruthy();
  });

  it('should be able to click the component one time', () => {
    const onClick = jest.fn();

    makeSut({ onClick, content });

    const repositoryCard = document.querySelector('.repositoryCard');
    user.click(repositoryCard);

    expect(onClick).toBeCalled();
  });

  it('should be able to click the component three times', () => {
    const onClick = jest.fn();

    makeSut({ onClick, content });

    const repositoryCard = document.querySelector('.repositoryCard');
    user.click(repositoryCard);
    user.click(repositoryCard);
    user.click(repositoryCard);

    expect(onClick).toBeCalledTimes(3);
  });

  it('should be able to render snapshot', () => {
    const component = renderer.create(
      <RepositoryCard
        content={<div>repository content</div>}
        onClick={() => jest.fn()}
      />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
