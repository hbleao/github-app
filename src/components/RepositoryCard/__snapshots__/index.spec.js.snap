// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoryCard should be able to render snapshot 1`] = `
<button
  className="repositoryCard"
  onClick={[Function]}
>
  <div>
    repository content
  </div>
  <svg
    className="repositoryCard__iconArrow"
    fill="none"
    height="12"
    viewBox="0 0 8 12"
    width="8"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8 10.5L3 6L8 1.5L6.60066 -4.44801e-08L6.67556e-07 6L6.60066 12L8 10.5Z"
    />
  </svg>
</button>
`;
